exopy_hqc_legacy/tasks/tasks/instr/apply_mag_field_task.py

```diff
@@ -59,41 +59,47 @@ def perform(self, target_value=None):
 
         driver = self.driver
         normal_end = True
-        if (abs(driver.read_persistent_field() - target_value) >
-                driver.output_fluctuations):
+        # Only if driver.heater_state == 'Off' otherwise we wait for 
+        # post_switch_wait no matter what
+        if driver.heater_state == 'Off':
             job = driver.sweep_to_persistent_field()
             if job.wait_for_completion(self.check_for_interruption,
-                                       timeout=60, refresh_time=1):
+                                       timeout=60, refresh_time=3):
                 driver.heater_state = 'On'
                 sleep(self.post_switch_wait)
             else:
                 return False
 
+        if (abs(driver.read_persistent_field() - target_value) > 
+            driver.output_fluctuations):
             # set the magnetic field
             job = driver.sweep_to_field(target_value, self.rate)
             normal_end = job.wait_for_completion(self.check_for_interruption,
                                                  timeout=60,
                                                  refresh_time=10)
+            print('field:', driver.read_persistent_field())
 
         # Always close the switch heater when the ramp was interrupted.
         if not normal_end:
-            job.cancel()
-            driver.heater_state = 'Off'
-            sleep(self.post_switch_wait)
+            job.cancel() # stops the sweep and turn off the switch heater
             self.write_in_database('field', driver.read_persistent_field())
-            return False
+            # if not normal_end, fail the measurement
+            self.root.should_stop.set()
+            raise ValueError(cleandoc('''Field source did not set the field to 
```

Comment:
This is debatable since it will appear as a hard failure even if the measurement was properly interrupted. This point to the need of differentiating between stop and timeout as I was suggesting in #61 .

Reply:
Mmm I was under the impression that when the measurement was manually stopped, the program stopped before going there but it is not clear to me how the stop signal is dealt with. 
I am fine with #61 !